blocks/header/header.js

```diff
@@ -126,41 +126,79 @@ function toggleMenu(nav, navSections, closeAll = null) {
   }
 }
 
-function getDirectTextContent(menuItem) {
-  const menuLink = menuItem.querySelector(':scope > a');
-  if (menuLink) {
-    return menuLink.textContent.trim();
-  }
-  return Array.from(menuItem.childNodes)
-    .filter((n) => n.nodeType === Node.TEXT_NODE)
-    .map((n) => n.textContent)
-    .join(' ');
-}
-
-async function buildBreadcrumbsFromNavTree(nav, currentUrl) {
+// function getDirectTextContent(menuItem) {
+//   const menuLink = menuItem.querySelector(':scope > a');
+//   if (menuLink) {
+//     return menuLink.textContent.trim();
+//   }
+//   return Array.from(menuItem.childNodes)
+//     .filter((n) => n.nodeType === Node.TEXT_NODE)
+//     .map((n) => n.textContent)
+//     .join(' ');
+// }
+
+// async function buildBreadcrumbsFromNavTree(nav, currentUrl) {
+//   const crumbs = [];
+
+//   const homeUrl = document.querySelector('.nav-brand a').href;
+
+//   let menuItem = Array.from(nav.querySelectorAll('a')).find((a) => a.href === currentUrl);
+//   if (menuItem) {
+//     do {
+//       const link = menuItem.querySelector(':scope > a');
+//       crumbs.unshift({ title: getDirectTextContent(menuItem), url: link ? link.href : null });
+//       menuItem = menuItem.closest('ul')?.closest('li');
+//     } while (menuItem);
+//   } else if (currentUrl !== homeUrl) {
+//     crumbs.unshift({ title: getMetadata('og:title'), url: currentUrl });
+//   }
+
+//   const homePlaceholder = 'Home';
+//   crumbs.unshift({ title: homePlaceholder, url: homeUrl });
+
+//   // last link is current page and should not be linked
+//   if (crumbs.length > 1) {
+//     crumbs[crumbs.length - 1].url = null;
+//   }
+//   crumbs[crumbs.length - 1]['aria-current'] = 'page';
+//   return crumbs;
+// }
+
+async function buildBreadcrumbsFromUrl(currentUrl) {
   const crumbs = [];
-
   const homeUrl = document.querySelector('.nav-brand a').href;
+  const urlObj = new URL(currentUrl);
+  const pathSegments = urlObj.pathname.split('/').filter(Boolean); // Remove empty strings from the array
 
-  let menuItem = Array.from(nav.querySelectorAll('a')).find((a) => a.href === currentUrl);
-  if (menuItem) {
-    do {
-      const link = menuItem.querySelector(':scope > a');
-      crumbs.unshift({ title: getDirectTextContent(menuItem), url: link ? link.href : null });
-      menuItem = menuItem.closest('ul')?.closest('li');
-    } while (menuItem);
-  } else if (currentUrl !== homeUrl) {
-    crumbs.unshift({ title: getMetadata('og:title'), url: currentUrl });
-  }
+  // Always add the home breadcrumb
+  crumbs.push({ title: 'Home', url: homeUrl });
```

Comment:
Does this ensure that we do **not** show a breadcrumb on the homepage?  (The comment "Always add the home breadcrumb" has me worried. 😸 )

Reply:
This just adds the "Home" element to the breadcrumbs chain. Home page will not have breadcumbs because it does not have the appropriate block in document